### DISCUSSION TITLE
Relations, SQL querie, and Despair

### DISCUSSION THREAD

**Discussion by EvanGllt** [Budibase/budibase#14113] [Help] [answered]
***Translated with deepl***

Hello there!

I have several problems with BUdibase that I can't solve, including :

- I can't make an SQL query to specifically get certain data from my table without it skipping the specifically selected row, at the mouse, always returning the same row. This happens as soon as I select "Queries" that I've created, instead of directly providing the raw table.
Example:
line 1 = first name: john, last name: doe
line 2 = first name: john, last name: doe)
Expected result:
line 1 = first name: john, last name: doe
line 2 = first name: Alan, last name: Turing

![aide 1](https://github.com/Budibase/budibase/assets/173894293/bb6cb7c9-9f66-4952-b63b-56e1dd90b770)
![aide 2](https://github.com/Budibase/budibase/assets/173894293/bcb0e667-5d91-4387-8201-07e3ae5a6730)
![aide 3](https://github.com/Budibase/budibase/assets/173894293/7ea420be-e83d-4ebd-97cb-c7f9045f96f5)

- I can't make an SQL query to fill the side panel with the same data as in the previously clicked table

- how can I click on a row in a table to redirect to the related row(s)?

I've had a look at the budibase documentation and can't find anything, except that it doesn't work.

Can you help?
Thanks in advance.

**Comment by mikesealey** [comment, marked as answer]
Hey @EvanGllt 

For your first issue, it looks like you need to specify the Row ID in the settings of the form in your Form Block component. 
![image](https://github.com/Budibase/budibase/assets/39269380/9f05303b-bdf0-4358-97fd-001842e8cced)

Without this, it will get _every row_, and only be able to show you the first one. You can do this by adding am "On row click" action to set the id of the row in state. 
![image](https://github.com/Budibase/budibase/assets/39269380/3a7fa4dc-d107-48ae-8055-b8e04d9dec23)
 
 and then use that value for the Row ID in the form. 
![image](https://github.com/Budibase/budibase/assets/39269380/d1a9ae8b-7d1c-41ba-bd16-512f2fdc70ac)

This should now open the form with the details of the clicked row as expected.

.

> I can't make an SQL query to fill the side panel with the same data as in the previously clicked table
you wouldn't even need to make an additional query, if you have the data in the table you could simple Set State with the Clicked Row binding
![image](https://github.com/Budibase/budibase/assets/39269380/2a5574b1-a3ac-4de5-aa0c-d7952db4e65c)

.

> how can I click on a row in a table to redirect to the related row(s)?

If you want to redirect the user to a page specific to the clicked row you could use this guide as a foundation https://docs.budibase.com/docs/passing-bindings-in-url-parameters

**Reply by EvanGllt** [reply]
Thanks for your help, but it still doesn't work for the first problem 😭

Here is exactly what I have:

- my components :
![aide 1](https://github.com/Budibase/budibase/assets/173894293/a03000ca-7819-4953-96aa-4c28e1d4f23a)

- my table (using a SQL Query) + action :
![aide 2](https://github.com/Budibase/budibase/assets/173894293/b1e5f371-9ba1-4847-bfd6-99992d9186a6)
![aide 3](https://github.com/Budibase/budibase/assets/173894293/2c0b0b07-31e7-4189-b493-5f94c9269cea)

- my SQL query : 
![aide 4](https://github.com/Budibase/budibase/assets/173894293/eb98bdce-8222-4803-82ea-7cb302fd5b59)

- my Form Block component :
![aide 4](https://github.com/Budibase/budibase/assets/173894293/e6d1a74d-6a69-4b98-a312-fe95d619502a)

**Reply by EvanGllt** [reply]
OK, I'm an idiot, I didn't put xxx_user_pk_id. 😅
Thanks a lot! 👌
